### PR TITLE
Expanded on Python I/O

### PR DESCRIPTION
**Pull request by DingleBells**
Added file reading using .readlines() and .readline(), writing to files using .write(), and basic tutorial on f-strings.

### CODE REVIEW

File: content/1_General/IO.mdx [hunk 1]
```diff
@@ -747,6 +747,54 @@ import sys
...
+A different approach to file I/O in Python is to still use the `open()` method, 
+but use the built-in functions `.readline()` or `.readlines()`. In Python, 
+`.readline()` will take the next line and return it as a string. This method 
+is useful for problems where you only have to read a short amount of lines
+but you still need to map each value to a variable. The `.readlines()` method 
+reads the entire file and returns a list of the file. Combined with a for loop, 
+this method provides a concise way to separate variables in the same line in a 
+problem. To write to the output file, you can use the `.write()` method. A 
+convienent way to convert a non-string value to a string value is to use 
+something called an f-string. An f-string is a really useful way to have
```
Comment (nwatx) [inline content/1_General/IO.mdx:761]:
```suggestion
an f-string. An f-string is a really useful way to have 
```

**Review by nwatx — APPROVED** [review]

**Comment by DingleBells** [comment]
@nwatx can you merge or do you have to wait for Nathan?

**Comment by nwatx** [comment]
the latter

**Comment by DingleBells** [comment]
ah ok

**Pull request merged**